File: leetcode/go/easy/001.go
func twoSum(nums []int, target int) []int {
    sum_map := make(map[int]int)

    for i, num := range nums {
        remainder := target - num

        if idx, ok := sum_map[remainder]; ok {
            return []int{idx, i}
        }

        sum_map[num] = i
    }

    return nil
}

File: leetcode/rust/easy/001.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut map = HashMap::new();
        
        for i in 0..nums.len() {
            let remainder = (target - nums[i]);
            
            if let Some(idx) = map.get(&remainder) {
                return vec![i as i32, *idx as i32];
            } else {
                map.insert(nums[i], i);
            }
        }
        
        Vec::new()
    }
}
